src/server/src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -570,32 +570,54 @@ public void handle(int repairNumber, Optional<ActiveRepairService.Status> status
       // See status explanations at: https://wiki.apache.org/cassandra/RepairAsyncAPI
-      // Old repair API
+      // Old repair API – up to Cassandra-2.1.x
       if(status.isPresent()) {
         switch (status.get()) {
           case STARTED:
-            renewLead();
-            DateTime now = DateTime.now();
-            context.storage.updateRepairSegment(currentSegment.with()
-                .state(RepairSegment.State.RUNNING)
-                .startTime(now)
-                .build(segmentId));
-            LOG.debug("updated segment {} with state {}", segmentId, RepairSegment.State.RUNNING);
+            try {
+              if (renewLead()) {
+                DateTime now = DateTime.now();
+                
+                context.storage.updateRepairSegment(currentSegment.with()
+                    .state(RepairSegment.State.RUNNING)
+                    .startTime(now)
+                    .build(segmentId));
+                
+                LOG.debug("updated segment {} with state {}", segmentId, RepairSegment.State.RUNNING);
+                break;
+              }
+            } catch (AssertionError er) {
+              // ignore. segment repair has since timed out.
+            }
+            segmentFailed.set(true);
             break;
 
           case SESSION_SUCCESS:
-            if (timedOut.get()) {
-              LOG.debug("Got SESSION_SUCCESS for segment with id '{}' and repair number '{}', " +
-                        "but it had already timed out",
-                        segmentId, repairNumber);
-            } else {
-              LOG.debug("repair session succeeded for segment with id '{}' and repair number '{}'",
-                      segmentId, repairNumber);
-              context.storage.updateRepairSegment(currentSegment.with()
-                  .state(RepairSegment.State.DONE)
-                  .endTime(DateTime.now())
-                  .build(segmentId));
+            try {
+              if (segmentFailed.get()) {
+                LOG.debug(
+                    "Got SESSION_SUCCESS for segment with id '{}' and repair number '{}', but it had already timed out",
+                    segmentId, repairNumber);
+              } else if (renewLead()) {
+                try {
+                  LOG.debug("repair session succeeded for segment with id '{}' and repair number '{}'", segmentId,
+                      repairNumber);
+
+                  context.storage.updateRepairSegment(currentSegment.with()
+                      .state(RepairSegment.State.DONE)
+                      .endTime(DateTime.now())
+                      .build(segmentId));
+
+                  break;
+                }  finally {
+                  releaseLead();
+                }
+              }
+            } catch (AssertionError er) {
+              // ignore. segment repair has since timed out.
             }
+            // this breaks repairs, reason unknown
+            //timedOut.set(true);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)